### PR TITLE
Add new section for GitLab Self-Managed

### PR DESCRIPTION
**Pull request by YashodhaSamarawickrama**
Related issue - https://github.com/wso2-enterprise/choreo/issues/29600

### CODE REVIEW

File: en/docs/develop-components/develop-components-with-git.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Authorizing Choreo as a GitHub application provides Choreo the following permiss
 
 ## Authorize Bitbucket with Choreo
 
-Authorizing Choreo as a Bitbucket application provides Choreo the following permissions to perform the respective actions on your behalf within the repository:
+Authorizing using a Personal Access Token (PAT) from Bitbucket provides Choreo with the following permissions to perform the respective actions on your behalf within the repository.
```
Comment (Nashaath) [inline en/docs/develop-components/develop-components-with-git.md:42]: In this case we don't need to capitalize the spelled-out version of PAT. The style guide recommends capitalizing the spelled-out version of the abbreviation only if the long form is a proper noun or is conventionally capitalized [1].
[1] https://developers.google.com/style/abbreviations#spelling-out

File: en/docs/develop-components/develop-components-with-git.md [hunk 2]
```diff
@@ -48,6 +48,13 @@ Authorizing Choreo as a Bitbucket application provides Choreo the following perm
 |Pull Requests | Y   | Y    | Create a pull request if you have chosen to start with a Choreo sample |
 |Webhooks      | Y   | Y    | Trigger automatic deployment and configuration generation              |
 
+## Authorize GitLab Self-Managed with Choreo
```
Comment (Nashaath) [inline en/docs/develop-components/develop-components-with-git.md:51]: Shall we say " Authorize Self-Managed GitLab with Choreo"

File: en/docs/develop-components/develop-components-with-git.md [hunk 3]
```diff
@@ -48,6 +48,13 @@ Authorizing Choreo as a Bitbucket application provides Choreo the following perm
...
 
+Authorizing using a Personal Access Token (PAT) obtained from your GitLab Self-Managed Server provides Choreo with the following permissions to perform the respective actions on your behalf within the repository.
```
Comment (Nashaath) [inline en/docs/develop-components/develop-components-with-git.md:53]: Same suggestion as above on capitalization.

File: en/docs/develop-components/develop-components-with-git.md [hunk 4]
```diff
@@ -48,6 +48,13 @@ Authorizing Choreo as a Bitbucket application provides Choreo the following perm
...
+
+|Permission    | Description                                                                         |
+|--------------|-------------------------------------------------------------------------------------|
+|API           | Grants full read/write access to the API, covering all groups and projects, 
+               | as well as read/write access to the repository.
```
Comment (Nashaath) [inline en/docs/develop-components/develop-components-with-git.md:58]: Might need to fix the table formatting here:

|Permission    | Description                                                                         |
|--------------|-------------------------------------------------------------------------------------|
|API           | Grants full read/write access to the API, covering all groups and projects, as well as read/write access to the repository.|

Comment (YashodhaSamarawickrama) [reply en/docs/develop-components/develop-components-with-git.md]: Addressed in https://github.com/YashodhaSamarawickrama/docs-choreo-dev/pull/1

File: en/docs/develop-components/develop-components-with-git.md (hunk 3 above)
Comment (YashodhaSamarawickrama) [reply en/docs/develop-components/develop-components-with-git.md]: Addressed in https://github.com/YashodhaSamarawickrama/docs-choreo-dev/pull/1

File: en/docs/develop-components/develop-components-with-git.md (hunk 2 above)
Comment (YashodhaSamarawickrama) [reply en/docs/develop-components/develop-components-with-git.md]: Addressed in https://github.com/YashodhaSamarawickrama/docs-choreo-dev/pull/1

File: en/docs/develop-components/develop-components-with-git.md (hunk 1 above)
Comment (YashodhaSamarawickrama) [reply en/docs/develop-components/develop-components-with-git.md]: Addressed in https://github.com/YashodhaSamarawickrama/docs-choreo-dev/pull/1

**Review by Nashaath — APPROVED** [review]

**Pull request merged**